### PR TITLE
initial commit for enhancing fuzz tests

### PR DESCRIPTION
**Pull request by sepehrdaddev**
Adds multiple fuzz targets with new directory structure and makefile.

libfuzzer and sanitizer options mirror oss-fuzz.

### CODE REVIEW

File: fuzz/fuzz_targets/FuzzServer.go [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+		cfg.ApplicationLog = "/dev/null"
+
+		go func() {
+			log.Fatal(skipper.Run(cfg.ToOptions()))
+		}()
+
+		initialized = true
+	}
+
+	conn, err := net.Dial("tcp", "localhost:9090")
```
[the remark this answers is not in the document]
Comment (sepehrdaddev) [reply fuzz/fuzz_targets/FuzzServer.go]: I have added a retry mechanism as per your suggestion.

File: fuzz/fuzz_targets/FuzzServer.go [hunk 2]
```diff
@@ -31,20 +46,29 @@ func connect(host string) (net.Conn, error) {
...
 		go func() {
 			log.Fatal(skipper.Run(cfg.ToOptions()))
 		}()
 
 		initialized = true
+		address = cfg.Address
 	}
 
-	conn, err := connect("localhost:9090")
+	conn, err := connect(address)
```
Comment (szuecs) [inline fuzz/fuzz_targets/FuzzServer.go:71]: you need to do something like:

```
var (
    err error
    conn *tcp.Conn // out of my head so maybe other type
)
for i:=0; i<10; i++ {
    conn, err = connect(address)
    if err != nil {
        time.Sleep(100*time.Millisecond)
        continue
    } 
    break
}
```

**Comment by szuecs** [comment]
:+1:

**Comment by sepehrdaddev** [comment]
:+1:

**Comment by sepehrdaddev** [comment]
:+1:

**Comment by sepehrdaddev** [comment]
@szuecs I will create follow up PRs to add these fuzz targets to oss-fuzz and then another one to remove the old ones from `eskip` directory.

**Pull request merged**